.github/workflows/build-and-test.yml

```diff
@@ -154,71 +154,55 @@ jobs:
+  build-wasm:
     needs: check-for-run-condition
     if: ${{ needs.check-for-run-condition.outputs.should-run == 'true' }}
+    # owner/repository of workflow has to be static, see https://github.community/t/env-variables-in-uses/17466
+    uses: iotaledger/identity.rs/.github/workflows/shared-build-wasm.yml@chore/unifiy-ci #TODO change to dev
+    with:
+      output-artifact-name: identity-wasm-bindings-build
+
+  test-wasm:
+    needs: build-wasm
+    if: ${{ needs.check-for-run-condition.outputs.should-run == 'true' }}
     runs-on: ubuntu-latest
     strategy:
       fail-fast: false
       matrix:
         os: [ ubuntu-latest ]
         include:
           - os: ubuntu-latest
-            sccache-path: /home/runner/.cache/sccache
-    env:
-      SCCACHE_DIR: ${{ matrix.sccache-path }}
-      RUSTC_WRAPPER: sccache
 
     steps:
       - uses: actions/checkout@v2
-
-      - uses: './.github/actions/rust/rust-setup'
-        name: rust-setup
-        with:
-          os: ${{ runner.os }}
-          job: ${{ github.job }}
-          target-cache-enabled: true
-          sccache-enabled: true
-          sccache-path: ${{ matrix.sccache-path }}
-          target-cache-path: bindings/wasm/target
-
-      - name: Install WASM toolchain
-        uses: actions-rs/toolchain@v1
-        with:
-          toolchain: stable
-          target: wasm32-unknown-unknown
-
-      - name: Setup sccache
-        uses: './.github/actions/rust/sccache/setup-sccache'
-        with:
-          os: ${{matrix.os}}
-
+        
       - name: Set up Node.js
         uses: actions/setup-node@v1
         with:
           node-version: 16.x
 
-      # Download a pre-compiled wasm-bindgen binary.
-      - name: Install wasm-bindgen-cli
-        uses: jetli/wasm-bindgen-action@24ba6f9fff570246106ac3f80f35185600c3f6c9
-
       - name: Install JS dependencies
-        run: npm install
-        working-directory: bindings/wasm
-
-      - name: Build WASM bindings
-        run: npm run build
+        run: npm ci
         working-directory: bindings/wasm
 
-      - name: Build Wasm examples
-        run: npm run build:examples
+      - name: Download bindings/wasm artifacts
+        uses: actions/download-artifact@v2
+        with:
+          name: identity-wasm-bindings-build
+          path: bindings/wasm
+    
+      - name: Run Wasm examples
+        run: npm run test:examples
         working-directory: bindings/wasm
 
-      - name: Run Wasm unit tests and examples
-        run: npm run test
-        working-directory: bindings/wasm
+  build-and-test-stronghold-nodejs:
+    needs: 
+      - check-for-run-condition
+      - build-wasm
+    if: ${{ needs.check-for-run-condition.outputs.should-run == 'true' }}
+    # owner/repository of workflow has to be static, see https://github.community/t/env-variables-in-uses/17466
+    uses: iotaledger/identity.rs/.github/workflows/shared-build-and-test-stronghold-nodejs.yml@chore/unifiy-ci #TODO change to dev
```

Comment:
TODO: needs an issue, don't forget.

Reply:
It is mentioned in the TODOs of the epic branch: https://github.com/iotaledger/identity.rs/pull/681